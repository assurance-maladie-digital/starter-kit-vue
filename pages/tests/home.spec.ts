// @vitest-environment nuxt
import Home from '../home.vue';
import NotFound from '../not-found.vue';
import { mountSuspended } from '@nuxt/test-utils/runtime';
import { describe, it, expect, vi } from 'vitest';
import { vuetify } from '@/tests/unit/setup';
import { mount } from '@vue/test-utils';
import {PageContainer, NotFoundPage} from "@cnamts/synapse-bridge";

describe('home page should render', () => {
	const mockRouter = {
		push: vi.fn()
	};

	const mockRoute = {
		path: '/some-path'
	};

	it('should render', async () => {
		const wrapper = await mountSuspended(Home,{
			global: {
				plugins: [
					vuetify,
				],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();

		const h1Text = wrapper.find('h1').text();
		expect(h1Text).toContain('Accueil');
	});

	it('should call setSupportId method', async () => {
		const wrapper = mount(NotFound,{
			global: {
				plugins: [
					vuetify,
				],
				mocks: {
					PageContainer,
					NotFoundPage,
					$router: mockRouter,
					$route: mockRoute
				},
			},
		});
		wrapper.vm.setSupportId();
		await wrapper.vm.$nextTick();

		expect(mockRouter.push).toHaveBeenCalled();
	});
});
