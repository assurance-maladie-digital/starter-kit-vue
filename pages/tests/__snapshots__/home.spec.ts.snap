// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home page should render > should render 1`] = `
"<h1 class="text-primary">Accueil</h1>
<div class="mt-6 mb-12 d-flex align-center justify-center">
  <div class="v-card v-theme--light v-card--density-default v-card--variant-tonal d-flex" style="width: 695px;">
    <!---->
    <div class="v-card__loader">
      <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
        <!---->
        <div class="v-progress-linear__background" style="width: 100%;"></div>
        <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!----><img src="/starter-kit.svg" alt="Starter kit">
    <!---->
    <!----><span class="v-card__underlay"></span>
  </div>
</div>
<div class="v-row justify-center">
  <div class="v-col-md-6 v-col-12">
    <div class="v-card v-theme--light bg-grey-lighten-4 v-card--density-default v-card--variant-elevated mx-auto">
      <!---->
      <div class="v-card__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="width: 100%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class="v-card-title"> Exemple de composant Nuxt </div>
      <div class="v-card-item">
        <!---->
        <div class="v-card-item__content">
          <!---->
          <!---->
          <div msg="Bonjour">
            <h1 class="text-primary">Hello World!</h1>
          </div>
        </div>
        <!---->
      </div>
      <!---->
      <!----><span class="v-card__underlay"></span>
    </div>
  </div>
  <div class="v-col-md-6 v-col-12">
    <div class="v-card v-theme--light bg-grey-lighten-4 v-card--density-default v-card--variant-elevated mx-auto">
      <!---->
      <div class="v-card__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="width: 100%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class="v-card-title"> Exemple de composant DS </div>
      <div class="v-card-item">
        <!---->
        <div class="v-card-item__content">
          <!---->
          <!----><a data-v-79141d7a="" class="v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text px-0 pr-1 vd-back-btn text-none" outlined="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i data-v-79141d7a="" class="v-icon notranslate v-theme--light v-icon--size-default text-primary mr-1" aria-hidden="true"><svg class="v-icon__svg" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true"><path d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"></path></svg></i>Retour</span>
            <!---->
            <!---->
          </a><br><br><a href="https://digital-design-system.netlify.app/demarrer/introduction/" target="_blank"> Voir la documentation des composants du Design System. </a>
        </div>
        <!---->
      </div>
      <!---->
      <!----><span class="v-card__underlay"></span>
    </div>
  </div>
</div>"
`;
